src/compas/datastructures/_mixins/attributes.py

```diff
@@ -768,6 +768,67 @@ def get_faces_attributes(self, keys, names, values=None):
         temp = list(zip(names, values))
         return [[self.get_face_attribute(fkey, name, value) for name, value in temp] for fkey in fkeys]
 
+    def get_face_all_attributes(self, key, data=True):
+        """Get all attributes of one face.
+
+        Parameters
+        ----------
+        key : hashable
+            The identifier of the face.
+        data : bool, optional
+            Returns the attributes and their values.
+            Default is ``True``.
+
+        Returns
+        ------
+        dict
+            The dictionary containing all attributes and their values, if ``data=True``.
+        list
+            A list of all attribute keys, if ``data=False`.
+
+        See Also
+        --------
+        * :meth:`get_face_attribute`
+        * :meth:`get_face_attributes`
+        * :meth:`get_faces_attribute`
+        * :meth:`get_faces_attributes`
+        * :meth:'get_faces_all_attributes'
+
+        """
+        if data:
+            return self.facedata[key]
+        else:

```

Comment:
should this be an editable dict or a copy?

Reply:
Whatever is best. It can also return a copy.

Reply:
perhaps it could be an option?